### PR TITLE
Add JS styleguide

### PR DESCRIPTION
**Pull request by arielsvn**
#627

### CODE REVIEW

**Comment by arielsvn** [comment]
@dongbohu @vincerubinetti can you help me review this? It's a bit long but it's mostly syntax changes to comply with [Airbnb's style guide](http://airbnb.io/javascript/).

**Comment by dongbohu** [comment]
@arielsvn I will try to do it tomorrow.

File: src/components/Strings.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
 export function NSamples({ total }) {
   if (!total) {
     return 'No Samples';
-  } else if (total === 1) {
+  }
+  if (total === 1) {
```
Comment (dongbohu) [inline src/components/Strings.js]: Could `total` be 0?

File: src/components/Strings.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
   }
+  return `${total} Samples`;
 }
 
 export function NDownloadableSamples({ total }) {
   if (!total) {
     return 'No Downloadable Samples';
-  } else if (total === 1) {
+  }
+  if (total === 1) {
```
Comment (dongbohu) [inline src/components/Strings.js]: Same question: could `total` be zero?

File: src/common/helpers.js [hunk 3]
```diff
@@ -268,14 +261,14 @@ export function truncateOnWord(str, limit, end = '...') {
...
-  var a = document.createElement('a');
+  const a = document.createElement('a');
   a.href = str;
   return a.host && a.host !== window.location.host;
 }
 
 function accumulate(array, sum) {
-  let result = [array[0]];
-  for (let i = 1; i < array.length; i++) {
+  const result = [array[0]];
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply src/common/helpers.js]: Nice catch, will check for that with this:

```suggestion
  if (!array || array.length === 0) return [];
  const result = [array[0]];
```

File: src/components/HighlightedText.js [hunk 4]
```diff
@@ -8,16 +8,15 @@ export default function HighlightedText({ text, highlight }) {
...
-        <span key={i} className="text-highlight">
-          {part}
-        </span>
-      ) : (
-        <React.Fragment key={i}>{part}</React.Fragment>
-      )
+  const parts = text.split(new RegExp(`(${highlight})`, 'gi'));
+  return parts.map(part =>
+    part && part.toLowerCase() === highlight.toLowerCase() ? (
+      <span key={part} className="text-highlight">
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply src/components/HighlightedText.js]: good catch, `part` doesn't have to be unique within `parts`. Will fix it!

File: src/components/Loader/index.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
-import { REPORT_ERROR } from '../../state/reportError';
 import isEqual from 'lodash/isEqual';
+import { REPORT_ERROR } from '../../state/reportError';
 
-// Component abtract the logic of loading content from the server before displaying something,
+// Component abtract the logic of loading content from the server before
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply src/components/Loader/index.js]:
```suggestion
// Component that abstracts the logic of loading content from the server before
```

File: src/components/Pagination/index.js [hunk 6]
```diff
@@ -39,10 +40,11 @@ class Pagination extends Component {
             )}
           </span>
         )}
-        {pageArray.map((page, i) => {
+        {pageArray.map(page => {
           return (
             <button
-              key={i}
+              key={page}
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply src/components/Pagination/index.js]: Yes, in this case `page` should be a number that is unique within `pageArray`

File: src/components/Strings.js (hunk 1 above)
Comment (arielsvn) [reply src/components/Strings.js]: yes, and in that case this should return `'No Samples'`

File: src/components/Strings.js (hunk 2 above)
Comment (arielsvn) [reply src/components/Strings.js]: same, for that case this should return 'No Downloadable Samples'

File: src/components/Strings.js (hunk 1 above)
Comment (dongbohu) [reply src/components/Strings.js]: Thanks. I didn't see the first `if` block.

File: src/components/Strings.js (hunk 2 above)
Comment (dongbohu) [reply src/components/Strings.js]: :+1:

**Comment by arielsvn** [comment]
@dongbohu thanks a lot for reviewing, this was a big PR. I pushed a few [additional changes](https://github.com/AlexsLemonade/refinebio-frontend/pull/640/files/d088afdf6f082111cc69d3752d39bc66aa3adda4..e1252e525bd93b45731117b964b808678df2ae6e) to ensure `key` values were unique. Let me know if you want to take another look.

**Pull request merged**